src/hotspot/share/opto/library_call.cpp

```diff
@@ -5206,24 +5206,108 @@ bool LibraryCallKit::inline_vectorizedMismatch() {
   const TypeAryPtr* top_a = a_type->isa_aryptr();
   const TypeAryPtr* top_b = b_type->isa_aryptr();
   if (top_a == NULL || top_a->klass() == NULL ||
-    top_b == NULL || top_b->klass() == NULL) {
+      top_b == NULL || top_b->klass() == NULL) {
     // failed array check
     return false;
   }
-
-  Node* call;
   jvms()->set_should_reexecute(true);
 
   Node* obja_adr = make_unsafe_address(obja, aoffset);
   Node* objb_adr = make_unsafe_address(objb, boffset);
 
-  call = make_runtime_call(RC_LEAF,
-    OptoRuntime::vectorizedMismatch_Type(),
-    stubAddr, stubName, TypePtr::BOTTOM,
-    obja_adr, objb_adr, length, scale);
+  assert(scale->bottom_type()->isa_int() &&
+         scale->bottom_type()->is_int()->is_con(),
+         "non-constant scale value");
+
+  int scale_val = scale->bottom_type()->is_int()->get_con();
+  BasicType prim_types[] = {T_BYTE, T_SHORT, T_INT, T_LONG};
+  BasicType vec_basictype = prim_types[scale_val];
+  const Type* vec_type = Type::get_const_basic_type(vec_basictype);
```

Comment:
It's not a vector type, but the element type. 

Also, should `VectorMaskGenNode` just accept element basic type instead?

Reply:
VectorMaskGenNode is of LONG type, explicit lane element type associated with VectorMaskGenNode is used to convert Store/LoadVectorMasked to Store/LoadVector for all true masks. Existing node definition expects a Type instead of a BasicType. I can change the name to vec_lane_type is that's acceptable.

Reply:
>  Existing node definition expects a Type instead of a BasicType.

Yes, what I suggested is to reconsider that. Unless I'm missing something important, a BasicType can be cached instead.